soc/microchip/pic32cxsg/common/Kconfig.pic32cxsg

```
@@ -0,0 +1,37 @@
+# Copyright (c) 2024 Microchip
+
+# SPDX-License-Identifier: Apache-2.0
+
+if SOC_FAMILY_MICROCHIP_PIC32CXSG
+
+config SOC_MICROCHIP_PIC32CXSG_XOSC32K
+	bool "The external 32 kHz crystal oscillator"
+	depends on DT_HAS_ATMEL_SAM0_OSC32KCTRL_ENABLED
+	help
+	  Say y to enable the external 32 kHZ crystal oscillator at
+	  startup. This can then be selected as the main clock source
+	  for the SOC.
+
+choice
+	prompt "Main clock source"
+	default SOC_MICROCHIP_PIC32CXSG_DEFAULT_AS_MAIN
+	help
+	  Selects the clock that the main clocks, such as the CPU
+	  clock and AHB clock, will be derived from.
+
+config SOC_MICROCHIP_PIC32CXSG_DEFAULT_AS_MAIN
+	bool "DEFAULT"
+	help
+	  This choice will leave all clocks to their current state.
+	  This can be the default reset state or a state set by a bootloader.
+
+config SOC_MICROCHIP_PIC32CXSG_XOSC32K_AS_MAIN
+	depends on SOC_MICROCHIP_PIC32CXSG_XOSC32K
+	bool "XOSC32K"
```

type comes first in Kconfigs e.g.
```suggestion
config SOC_MICROCHIP_PIC32CXSG_XOSC32K_AS_MAIN
	bool "XOSC32K"
	depends on SOC_MICROCHIP_PIC32CXSG_XOSC32K
```

---

@nordicjm Your request has been addressed, Please mark as Resolved. Thanks.

---

Updated Kconfig.pic32cxsg and removed the ones you pointed out. The remaining setting is set to y if osc32kctrl DT node is enabled.